drivers/media/i2c/lt6911uxc.c

```diff
@@ -0,0 +1,989 @@
+
+	dev_info(&client->dev, "Chip found @ 7h%02X (%s)\n", client->addr,
+		 client->adapter->name);
+
+	/* initial setup */
+	lt6911uxc_initial_setup(state);
+
+	/* get interrupt */
+	if (client->irq) {
+		err = devm_request_threaded_irq(&state->i2c_client->dev,
+						client->irq, NULL,
+						lt6911uxc_irq_handler,
+						IRQF_TRIGGER_FALLING |
+						IRQF_ONESHOT, sd->name,
+						(void *)sd);
+		if (err) {
+			dev_err(&client->dev, "Could not request interrupt %d!\n",
+				client->irq);
+			return err;
+		}
+	} else {
+		INIT_WORK(&state->work_i2c_poll, lt6911uxc_work_i2c_poll);
+		timer_setup(&state->timer, lt6911uxc_irq_poll_timer, 0);
+		state->timer.expires = jiffies + msecs_to_jiffies(POLL_INTERVAL_MS);
+		add_timer(&state->timer);
+	}
+
+	/* custom v4l2 controls */
+	v4l2_ctrl_handler_init(&state->ctrl_handler, 2);
+	state->audio_sampling_rate_ctrl = v4l2_ctrl_new_custom(
+		&state->ctrl_handler, &lt6911uxc_ctrl_audio_sampling_rate, NULL);
+	state->audio_present_ctrl = v4l2_ctrl_new_custom(&state->ctrl_handler,
+		&lt6911uxc_ctrl_audio_present, NULL);
+
+	v4l2_ctrl_handler_setup(sd->ctrl_handler);
+	if (state->ctrl_handler.error) {
+		err = state->ctrl_handler.error;
+		goto err_ctrl_handler;
+	}
+	sd->ctrl_handler = &state->ctrl_handler;
+
+	/* media entitiy: define pad as output -> origins of link */
+	state->pad.flags = MEDIA_PAD_FL_SOURCE;
+	sd->entity.ops = &lt6911uxc_media_ops;
+
+	err = media_entity_pads_init(&sd->entity, 1, &state->pad);
+	if (err < 0) {
+		dev_err(&client->dev, "unable to init media entity\n");
+		goto err_ctrl_handler;
+	}
+
+	/* register v4l2_subdev device */
+	sd->dev	= &client->dev;
+	sd->internal_ops = &lt6911uxc_subdev_internal_ops;
+	sd->flags |= V4L2_SUBDEV_FL_HAS_DEVNODE | V4L2_SUBDEV_FL_HAS_EVENTS;
+
+	err = v4l2_async_register_subdev(sd);
+	if (err) {
+		dev_err(&client->dev, "lt6911uxc subdev registration failed\n");
+		goto err_ctrl_handler;
+	}
+	return 0;
+
+ err_ctrl_handler:
+	v4l2_ctrl_handler_free(&state->ctrl_handler);
+	return err;
+}
+
+static int lt6911uxc_remove(struct i2c_client *client)
+{
+	struct v4l2_subdev *sd = i2c_get_clientdata(client);
+
+	v4l2_async_unregister_subdev(sd);
+	v4l2_ctrl_handler_free(sd->ctrl_handler);
+
+	if (IS_ENABLED(CONFIG_MEDIA_CONTROLLER))
+		media_entity_cleanup(&sd->entity);
+
+	dev_info(&client->dev, "removed lt6911uxc instance\n");
+	return 0;
+}
+
+static const struct i2c_device_id lt6911uxc_id[] = {
+	{ "lt6911uxc", 0 },
+	{ }
+};
+MODULE_DEVICE_TABLE(i2c, lt6911uxc_id);
+
+static struct i2c_driver lt6911uxc_driver = {
+	.driver = {
+		.of_match_table = of_match_ptr(lt6911uxc_of_match),
+		.name = "lt6911uxc",
+		.owner = THIS_MODULE,
+	},
+	.id_table = lt6911uxc_id,
+	.probe	  = lt6911uxc_probe,
+	.remove   = lt6911uxc_remove,
+};
+module_i2c_driver(lt6911uxc_driver);
+
+MODULE_DESCRIPTION("Driver for Lontium lt6911uxc HDMI to CSI-2 Bridge");
+MODULE_AUTHOR("Lukas Neuner <neur@zhaw.ch>");
+MODULE_AUTHOR("Alexey Gromov <groo@zhaw.ch>");
```

Comment:
Do they know that you're listing them as author. Any support requests are likely to head back to them if this is left here.